store/v2/utils.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2021 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package store
+
+import (
+	"context"
+	"strings"
+
+	"go.mongodb.org/mongo-driver/bson"
+
+	"github.com/mendersoftware/go-lib-micro/identity"
+	v1 "github.com/mendersoftware/go-lib-micro/store"
+)
+
+const FieldTenantID = "tenant_id"
+
+// WithTenantID adds the tenant_id field to a bson document using the value extracted
+// from the identity of the context
+func WithTenantID(ctx context.Context, doc interface{}) bson.D {
+	res := bson.D{}
+
+	identity := identity.FromContext(ctx)
+	if identity != nil {
+		res = append(res, bson.E{Key: FieldTenantID, Value: identity.Tenant})
+	}
+
+	switch v := doc.(type) {
+	case map[string]interface{}:
+		for k, v := range v {
+			res = append(res, bson.E{Key: k, Value: v})
+		}
+	case bson.M:
+		for k, v := range v {
+			res = append(res, bson.E{Key: k, Value: v})
+		}
+	case bson.D:
+		res = append(res, v...)
+	default:
+		if data, err := bson.Marshal(v); err == nil {
+			bsonData := bson.D{}
+			if err = bson.Unmarshal(data, &bsonData); err == nil {
+				res = append(res, bsonData...)
+			}
+		}
+	}
+
+	return res
+}
+
+// ArrayWithTenantID adds the tenant_id field to an array of bson documents
+// using the value extracted from the identity of the context
+func ArrayWithTenantID(ctx context.Context, doc bson.A) bson.A {
```

Comment:
I guess this could also be implemented as a case in `WithTenantID`?

Reply:
It was tricky because they have different return types. How could I do it?